pilot/pkg/kube/inject/inject.go

```diff
@@ -107,6 +107,11 @@ func InitImageName(hub string, tag string, _ bool) string {
 // ProxyImageName returns the fully qualified image name for the istio
 // proxy image given a docker hub and tag and whether to use debug or not.
 func ProxyImageName(hub string, tag string, debug bool) string {
+	// Allow overriding the proxy image.
+	image := os.Getenv("ISTIO_PROXY_IMAGE")
+	if image != "" {
+		return hub + "/" + image + ":" + tag
+	}
```

Comment:
why again?

Reply:
It's an easy way to customize the injected image - works with both webhook and command line.

The official solution will likely be an annotation - flags in istioclt are not going to work for webhook.

Reply:
no I meant why add this? the previous PR already had this right?